### PR TITLE
WRR-12000: Enhanced ColorPicker code and made component public

### PR DESCRIPTION
**Pull request by stanca-pop-lgp**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [x] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
ColorPickerPOC component needs improvements:
- fix any bugs
- find and perform any necessary refactoring
- write suitable JSDocs (make the component public)
- fix any typos 

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Performed enhancements on ColorPickerPOC component:
- renamed component to ColorPicker and made the primary component public
- added JS docs
- refactored code
- fixed any lint warnings

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRR-12000

### Comments
Enact-DCO-1.0-Signed-off-by: Daniel Stoian (daniel.stoian@lgepartner.com)

### CODE REVIEW

**Comment by stanca-pop-lgp** [comment]
Tests will be added in a separate ticket.

File: ColorPicker/ColorPickerSlider.js [hunk 1]
```diff
@@ -178,9 +201,37 @@ const ColorPickerSliderRGB = ({selectedColor, selectedColorHandler, ...props}) =
...
+	 */
+	selectedColorHandler: PropTypes.func
+};
+
+/**
+ * A color picker component that allows the user to choose a color by using an HSL color slider.
+ *
+ * @class ColorPickerSliderHSL
+ * @memberof sandstone/ColorPicker
+ * @extends sandstone/Slider.Slider
```
Comment (daniel-stoian-lgp) [inline ColorPicker/ColorPickerSlider.js:229]: on my opinion, a component can extend another component only if the main node is the other component or it is just wrapped in other HTML elements. In this case, I see the usage of Layout/Cell/Row

File: ColorPicker/ColorPickerSlider.js [hunk 2]
```diff
@@ -61,6 +75,15 @@ const lightnessGradient = (hue, saturation) => {
 	hsla(${hue}, ${saturation}%, 100%, 1))`;
 };
 
+/**
+ * A color picker component that allows the user to choose a color by using an RGB color slider.
+ *
+ * @class ColorPickerSliderRGB
+ * @memberof sandstone/ColorPicker
+ * @extends sandstone/Slider.Slider
```
Comment (daniel-stoian-lgp) [inline ColorPicker/ColorPickerSlider.js:83]: on my opinion, a component can extend another component only if the main node is the other component or it is just wrapped in other HTML elements. In this case, I see the usage of Layout/Cell/Row

File: ColorPicker/utils.js [hunk 3]
```diff
@@ -1,35 +1,143 @@
...
+ *
+ * @function
+ * @param {String} hexColor
+ * @returns {{h: number, s: number, l: number}}	HSL values
  * @private
  */
 const hexToHSL = (hexColor) => {
-	const hex = checkHex(hexColor);
 	// Convert hex to RGB first
 	let r = 0, g = 0, b = 0;
```
Comment (daniel-stoian-lgp) [inline ColorPicker/utils.js:137]: variable initiliazer is redundant. let's define r,g,b directly in the lines below

**Review by daniel-stoian-lgp — COMMENTED** [review]
check my comments

File: ColorPicker/ColorPickerSlider.js (hunk 1 above)
Comment (stanca-pop-lgp) [reply ColorPicker/ColorPickerSlider.js:229]: You are right. I will correct the docs.

**Review by daniel-stoian-lgp — APPROVED** [review]
LGTM

**Comment by adrian-cocoara-lgp** [comment]
Code coverage will be updated on a different PR: WRR-12002

File: ColorPicker/ColorPickerSlider.js [hunk 4]
```diff
@@ -1,14 +1,28 @@
+/**
+ * Sandstone component that allows the user to choose a color by using an RGB or an HSL color slider.
```
Comment (adrian-cocoara-lgp) [inline ColorPicker/ColorPickerSlider.js:2]:
```suggestion
 * Sandstone component that allows the user to choose a color by using a RGB or a HSL color slider.
```

File: ColorPicker/ColorPickerSlider.js [hunk 5]
```diff
@@ -61,7 +75,15 @@ const lightnessGradient = (hue, saturation) => {
 	hsla(${hue}, ${saturation}%, 100%, 1))`;
 };
 
-const ColorPickerSliderRGB = ({selectedColor, selectedColorHandler, ...props}) => {
+/**
+ * A color picker component that allows the user to choose a color by using an RGB color slider.
```
Comment (adrian-cocoara-lgp) [inline ColorPicker/ColorPickerSlider.js:79]:
```suggestion
 * A color picker component that allows the user to choose a color by using a RGB color slider.
```

File: ColorPicker/ColorPickerSlider.js [hunk 6]
```diff
@@ -178,9 +206,45 @@ const ColorPickerSliderRGB = ({selectedColor, selectedColorHandler, ...props}) =
...
+	 * Called when the selected color is modified.
+	 *
+	 * @type {Function}
+	 * @private
+	 */
+	selectedColorHandler: PropTypes.func
+};
+
+/**
+ * A color picker component that allows the user to choose a color by using an HSL color slider.
```
Comment (adrian-cocoara-lgp) [inline ColorPicker/ColorPickerSlider.js:239]:
```suggestion
 * A color picker component that allows the user to choose a color by using a HSL color slider.
```

File: ColorPicker/ColorPickerSpectrum.js [hunk 7]
```diff
@@ -20,7 +40,8 @@ const SpectrumColorPicker = (props) => {
 
 	useEffect(() => {
 		const canvas = canvasRef.current;
-		const ctx = canvas.getContext('2d');
+		const ctx = canvas.getContext('2d', {willReadFrequently: true});
+		// `willReadFrequently: true` parameter permits the browser to optimize for frequent getImageData() calls by avoiding hardware acceleration
```
Comment (adrian-cocoara-lgp) [inline ColorPicker/ColorPickerSpectrum.js:44]: this comment can be deleted and I can check it in a separate PR

File: ColorPicker/ColorPicker.js [hunk 8]
```diff
@@ -1,181 +1,244 @@
-/* eslint-disable react-hooks/rules-of-hooks */
 /**
- * Sandstone component to allow the user to choose a color.
+ * Sandstone component that allows the user to choose a color
+ * either from a grid, or a spectrum, or RGB/HSL color sliders.
```
Comment (adrian-cocoara-lgp) [inline ColorPicker/ColorPicker.js:3]:
```suggestion
 * either from a grid, a spectrum, or RGB/HSL color sliders.
```

File: ColorPicker/ColorPicker.js [hunk 9]
```diff
@@ -1,181 +1,244 @@
...
  *
  * @example
  * <ColorPicker
- *	 color={'#FF00FF'}
- *	 colorHandler={setColor}
- *	 presetColors={['#FF0000', '#00FF00', '#0000FF']}
- *	 text={'Color Picker'}
+ * 	 onChangeColor={console.log}
```
Comment (adrian-cocoara-lgp) [inline ColorPicker/ColorPicker.js:7]:
```suggestion
 * 	 onChangeColor={() => {}} or  onChangeColor={() => console.log}
```

File: ColorPicker/ColorPickerSlider.js (hunk 4 above)
Comment (stanca-pop-lgp) [reply ColorPicker/ColorPickerSlider.js:2]: "an" si also used before words that start with a vowel sound.

**Review by adrian-cocoara-lgp — APPROVED** [review]
LGTM

**Pull request merged**